### ISSUE TITLE
Assigning to `shape_layer.data` gives IndexError if shape layer is empty

### ISSUE THREAD

**Issue by VolkerH** [napari/napari#2960]
## 🐛 Bug

Assigning to `shape_layer.data` gives `IndexError` if shape layer is empty.
This seems to have been introduced with commit f7537e095d00b1c149455495374ad0cbbe3724f1
by @DragaDoncila.
I suspect the bug is here:
https://github.com/napari/napari/blob/350e07cc5861005d6bb3199f8e39e276f8d04b3c/napari/layers/shapes/_shapes_utils.py#L1030

## To Reproduce

1. Open napari Ipython console.
2. Run the following snippet

```
import numpy as np
box = np.array([[0,100],[100,100],[100,0],[0,0]])
s=viewer.add_shapes()
s.data = s.data + [box]
```

## Expected behavior

I would expect to be able to assign to `.data` of an empty shape layer.

## Environment

napari: 0.4.10
Platform: Linux-5.8.0-59-generic-x86_64-with-glibc2.31
System: Ubuntu 20.04.2 LTS
Python: 3.9.5 (default, Jun 4 2021, 12:28:51) [GCC 7.5.0]
Qt: 5.15.2
PyQt5: 5.15.4
NumPy: 1.21.0
SciPy: 1.7.0
Dask: 2021.06.2
VisPy: 0.6.6

OpenGL:
- GL version: 4.6.0 NVIDIA 460.80
- MAX_TEXTURE_SIZE: 32768

Screens:
- screen 1: resolution 3840x2160, scale 1.0

Plugins:
- console: 0.0.3
- dzi-zarr: 0.1.2
- scikit-image: 0.4.10
- svg: 0.1.5
- zarpaint: 0.1.0

**Comment by VolkerH** [comment]
I created a small bugfix PR. 
As it changes the default return type as well, someone should confirm whether this is the desired behaviour.

**Comment by alisterburt** [comment]
closed by #2961
